src/main/java/gregtech/api/metatileentity/TieredMetaTileEntity.java

```diff
@@ -68,6 +86,47 @@ public void renderMetaTileEntity(CCRenderState renderState, Matrix4 translation,
         getBaseRenderer().render(renderState, translation, colouredPipeline);
     }
 
+    @Override
+    public void update() {
+        super.update();
+        if (ConfigHolder.machines.doTerrainExplosion) {
+            checkTerrainExplosion();
+        }
+    }
+
+    public void checkTerrainExplosion() {
+        World world = getWorld();
+        if (!world.isRemote && !getTerrainResistance() && energyContainer.getEnergyStored() != 0) {
+            if (world.rand.nextInt(1000) == 0) {
+                for (EnumFacing side : EnumFacing.VALUES) {
+                    Block block = getWorld().getBlockState(getPos().offset(side)).getBlock();
+                    if (block == Blocks.FIRE || block == Blocks.WATER || block == Blocks.FLOWING_WATER || block == Blocks.LAVA || block == Blocks.FLOWING_LAVA) {
```

Comment:
Not sure if water should be included here, some discussion might be necessary on this one.